gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/IndexGigawordWithElasticSearch.java

```diff
@@ -1,52 +1,190 @@
-
-                // you can index documents one-by-one
-                client.index(new IndexRequest("documents", "doc", docId)
-                        .source("text", docContent));
-            }
-
-            // TODO: in practice, you will want to use BulkRequest for speed
+  private static final Logger log = LoggerFactory.getLogger(IndexGigawordWithElasticSearch.class);
+
+  private static final String DEFAULT_HOST = "localhost";
+
+  private static final int DEFAULT_PORT_PRI = 9200;
+
+  private static final int DEFAULT_PORT_SEC = 9201;
+
+  private static final String PARAM_GIGAWORD_FILEPATH = "gigawordFilePath";
+
+  private static final String PARAM_GIGAWORD_DIRECTORY_PATH = "gigawordDirectoryPath";
+
+  private static final String PARAM_INDEX_NAME = "indexName";
+
+  private static final String PARAM_HOSTNAME_PRIMARY = "primaryHostName";
+
+  private static final String PARAM_HOSTNAME_SECONDARY = "secondaryHostName";
+
+  private static final String PARAM_PORT_PRIMARY = "primaryPort";
+
+  private static final String PARAM_PORT_SECONDARY = "secondaryPort";
+
+  public static void main(String[] argv) throws IOException {
+
+    // get parameter file
+    Parameters parameters = null;
+
+    if (argv.length > 0) {
+      parameters = Parameters.loadSerifStyle(new File(argv[0]));
+    } else {
+      log.error("Expected one argument, a parameter file.");
+      System.exit(1);
+    }
+
+    try {
+
+      RestHighLevelClient client =
+          new RestHighLevelClient(
+              RestClient.builder(
+                  new HttpHost(
+                      parameters.getOptionalString(PARAM_HOSTNAME_PRIMARY).or(DEFAULT_HOST),
+                      parameters
+                          .getOptionalPositiveInteger(PARAM_PORT_PRIMARY)
+                          .or(DEFAULT_PORT_PRI),
+                      "http"),
+                  new HttpHost(
+                      parameters.getOptionalString(PARAM_HOSTNAME_SECONDARY).or(DEFAULT_HOST),
+                      parameters
+                          .getOptionalPositiveInteger(PARAM_PORT_SECONDARY)
+                          .or(DEFAULT_PORT_SEC),
+                      "http")));
+
+      String indexName = parameters.getString(PARAM_INDEX_NAME);
+
+      if (parameters.isPresent(PARAM_GIGAWORD_DIRECTORY_PATH)) {
+        File gigawordDir = parameters.getExistingDirectory(PARAM_GIGAWORD_DIRECTORY_PATH);
+        for (File concatenatedFileToIndex : gigawordDir.listFiles()) {
+          if (concatenatedFileToIndex.isFile() && isValidGzipFile(concatenatedFileToIndex)) {
+            index(client, concatenatedFileToIndex, indexName);
+          }
         }
+      } else {
+        File concatenatedFileToIndex = parameters.getExistingFile(PARAM_GIGAWORD_FILEPATH);
+        if (!isValidGzipFile(concatenatedFileToIndex)) {
+        	client.close();
+        	throw new Exception(concatenatedFileToIndex.getAbsolutePath() + " is not a valid GZip file");
```

Comment:
You want an `IOException` here - even if you want to throw an exception of unspecified type, prefer `RuntimeException` (because it is unchecked.  It doesn't matter much here in `main`, but it's good to be consistent)

Reply:
Rather than sprinkling `client.close()` in various branches, put it in a `finally` block

Reply:
Fixed. Please check.